cca_realm_claims.go

```diff
@@ -0,0 +1,242 @@
+// CCA Realm Claims
+package ccatoken
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/veraison/eat"
+)
+
+type CcaRealmClaims struct {
+	Challenge              *eat.Nonce `cbor:"10,keyasint" json:"cca-realm-challenge"`
+	PersonalizationValue   *[]byte    `cbor:"44235,keyasint" json:"cca-realm-personalization-value"`
+	InitialMeasurements    *[]byte    `cbor:"44238,keyasint" json:"cca-realm-initial-measurement"`
+	ExtensibleMeasurements *[][]byte  `cbor:"44239,keyasint" json:"cca-realm-extensible-measurements"`
+	HashAlgID              *string    `cbor:"44236,keyasint" json:"cca-realm-hash-algo-id"`
+	PubicKey               *[]byte    `cbor:"44237,keyasint" json:"cca-realm-public-key"`
+	PublickeyHashAlgID     *string    `cbor:"44240,keyasint" json:"cca-realm-public-key-hash-algo-id"`
+}
+
+// Setters
+
+func (c *CcaRealmClaims) SetChallenge(v []byte) error {
+	if err := isValidChallenge(v); err != nil {
```

Comment:
nit: "is" methods should return boolean values "is valid challenge?" is a yes/no answer. As this returns an error, I call it `checkValidChallenge` or simply `validateChallenge`. Similar below.

Reply:
This has been so far consistent with PSA token. However I agree and raised a github issue #4 to address it, when we do further sanity check for psatoken and cca token validation functions.